File: Code/new_fitervari_backend/src/main/java/at/htl/control/WorkoutHistoryRepository.java
package at.htl.control;

import at.htl.model.WorkoutHistory;
import io.quarkus.hibernate.orm.panache.PanacheRepository;

import javax.enterprise.context.ApplicationScoped;
import java.util.List;
import java.util.stream.Collectors;

@ApplicationScoped
public class WorkoutHistoryRepository implements PanacheRepository<WorkoutHistory> {

    public WorkoutHistory save(WorkoutHistory entity) {
        return getEntityManager().merge(entity);
    }

    public List<WorkoutHistory> getWorkoutHistoriesByWorkoutId(Long workoutId) {
       return streamAll()
               .filter(workoutHistory -> workoutHistory.getWorkout().getId() == workoutId)
               .collect(Collectors.toList());
    }
}

File: Code/fitervari_backend/src/main/java/at/htl_leonding/repository/WorkoutHistoryRepository.kt
package at.htl_leonding.repository

import at.htl_leonding.model.WorkoutHistory
import io.quarkus.hibernate.orm.panache.PanacheRepository
import javax.enterprise.context.ApplicationScoped
import javax.inject.Inject
import javax.persistence.EntityManager

@ApplicationScoped
class WorkoutHistoryRepository : PanacheRepository<WorkoutHistory> {
    @Inject
    lateinit var em: EntityManager

    fun save(workoutHistory: WorkoutHistory): WorkoutHistory {
        return em.merge(workoutHistory)
    }
}